tests/Linter/ProcessLintingResultTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Linter;
+
+use PhpCsFixer\Linter\ProcessLintingResult;
+use PhpCsFixer\Tests\TestCase;
+
+/**
+ * @author SpacePossum
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Linter\ProcessLintingResult
+ */
+final class ProcessLintingResultTest extends TestCase
+{
+    public function testCheckOK()
+    {
+        $process = $this->prophesize('Symfony\Component\Process\Process');
+        $process
+            ->wait()
+            ->willReturn(0)
+        ;
+
+        $process
+            ->isSuccessful()
+            ->willReturn(true)
+        ;
+
+        $result = new ProcessLintingResult($process->reveal());
+        $result->check();
+        $result->check(); // call twice
```

Comment:
why we are calling it twice ?

Reply:
`ProcessLintingResult` caches the result internal, so this tests checks that path as well (`if (null === $this->isSuccessful) {`)

Reply:
you are executing the path twice, but you are not testing was the path different.

if one would drop internal cache of the class, your tests would still pass, which means that you are not testing internal cache.

Reply:
removed